checker/src/org/checkerframework/checker/lock/LockTransfer.java

```diff
@@ -173,4 +196,18 @@ else if (astKind == Kind.ARBITRARY_CODE) { // Handle initializers
 
         return result;
     }
+
+    @Override
+    public TransferResult<CFValue, LockStore> visitStringConcatenate(
+            StringConcatenateNode n, TransferInput<CFValue, LockStore> p) {
+        TransferResult<CFValue, LockStore> result = super.visitStringConcatenate(n, p);
+
+        // Strings are always @GuardedBy({})
+
+        if (result.getResultValue().getType().hasAnnotation(GUARDEDBYUNKNOWN)) {
+            result.getResultValue().getType().replaceAnnotation(GUARDEDBY);
```

Comment:
This covers the following case. Indeed, the right operand would have type @GuardedBy({}) if it were de-sugared. Added a comment.

  // Tests that the resulting type of string concatenation is always @GuardedBy({})
  // (and not @GuardedByUnknown, which is the LUB of @GuardedBy({}) (the type of the
  // string literal "a") and @GuardSatisfied (the type of param))
  void StringConcat(/*@GuardSatisfied*/ MyClass param) {
      String s = "a" + param;
  }